### PR TITLE
Update GoodFET to resolve packet concatenation and fix the sync problem

### PR DESCRIPTION
**Pull request by inode-**
Using an old version of GoodFET caused problems in receiving ACK packets that were concatenated with data packets.

I made a patch for GoodFET which fixes some problems of the latest versions (https://github.com/travisgoodspeed/goodfet/pull/51) and compiled the current version which no longer has the chained ACK problem.

I also increased the "attemptslimit" parameter so that serial sync can be done correctly. An alternative is to do it manually with the command:

`goodfet/client/goodfet.monitor listapps full`

This patch is expected to fix several bugs, in particular:
https://github.com/riverloopsec/killerbee/issues/215
https://github.com/riverloopsec/killerbee/issues/192
https://github.com/riverloopsec/killerbee/issues/190
https://github.com/riverloopsec/killerbee/issues/166
https://github.com/riverloopsec/killerbee/issues/160
https://github.com/riverloopsec/killerbee/issues/159
https://github.com/riverloopsec/killerbee/issues/155

### CODE REVIEW

File: killerbee/GoodFET.py [hunk 1]
```diff
@@ -217,12 +216,12 @@ def pyserInit(self, port, timeout, attemptlimit):
                 elif (os.environ.get("board")=='z1'):
                     self.bslResetZ1(invokeBSL=0);
                 elif (os.environ.get("board")=='apimote1') or (os.environ.get("board")=='apimote'):
-                    #Not needed for the apimote2
```
Comment (rmspeers) [inline killerbee/GoodFET.py:220]: This is not needed for apimote2, 3, or 4 - only for apimote1, right?

File: killerbee/GoodFETCCSPI.py [hunk 2]
```diff
@@ -19,6 +19,7 @@ def setup(self):
         
         #Set up the radio for ZigBee
         self.strobe(0x01);       #SXOSCON
+        self.strobe(0x01);       #SXOSCON ## YL: idk why but needed to enable ram poking
```
Comment (rmspeers) [inline killerbee/GoodFETCCSPI.py:22]: @travisgoodspeed - this appears that this PR just updates code from GoodFET repo, do you have insight into this change as the comment makes me wonder about it? https://github.com/travisgoodspeed/goodfet/blob/master/client/GoodFETCCSPI.py#L22

File: killerbee/GoodFETCCSPI.py [hunk 3]
```diff
@@ -198,6 +197,20 @@ def pokeram(self,adr,data):
         return;
     
     lastpacket=range(0,0xff);
+
+    def RF_txrxpacket(self,packet,timeout=1):
```
Comment (rmspeers) [inline killerbee/GoodFETCCSPI.py:201]: Using this for zbstumbler type use cases may be helpful to get a potentially quicker RX to a forged beacon request? CC @taylorcenters

File: killerbee/GoodFET.py (hunk 1 above)
Comment (inode-) [reply killerbee/GoodFET.py:220]: All the GoodFET files has been directly token from the GoodFET repository. From my tests it's not necessary to update the python code, but due the new firmware I prefered to do it.

File: README.md [hunk 4]
```diff
@@ -98,6 +98,10 @@ The devices typically come preloaded and do not need to be reflashed for basic u
 The hardware is open-source at https://github.com/riverloopsec/apimote.
 It is available assembled by contacting team at riverloopsecurity dot com.
 
+In case of problems identifying the hardware try to run the following command (present in [GoodFET](https://github.com/travisgoodspeed/goodfet/blob/master/firmware/apps/radios/ccspi.c)) before any killerbee command
+
+`~/goodfet/client$ sudo ./goodfet.monitor listapps full`
+
```
Comment (rmspeers) [inline README.md:104]: We're going to move this to the FAQs if that's OK. @taylorcenters  is preparing the update off of this PR that includes that.

File: killerbee/scapy_extensions.py [hunk 5]
```diff
@@ -10,6 +10,7 @@
 from scapy.base_classes import SetGen
 from scapy.packet import Gen, Raw
 from scapy.all import *
+from scapy import plist
```
Comment (rmspeers) [inline killerbee/scapy_extensions.py:13]: Good fix!

**Review by rmspeers — COMMENTED** [review]
@inode- can you remind me what the purpose of the new FW update is? I think you mentioned at some point so I'm sorry if I forgot. We'd like to build it from source and be able to put the changes in release notes.

**Comment by inode-** [comment]
The new firmware use the latest version of GoodFET which doesn't concatenate to the data packets the ACK. This kind behavior was the cause of "Bad FCS" that a lot of people was having with ApiMote.

The other problem was that a lot of people need to detach and attach multiple time the ApiMote to get it working. Fixed in the  python code. Also the workaround was another way to "bypass" the sync problem.

Other minor fixes was done.

File: killerbee/scapy_extensions.py (hunk 5 above)
Comment (inode-) [reply killerbee/scapy_extensions.py:13]: Issue already reported and fixed at https://github.com/riverloopsec/killerbee/issues/207

**Review by taylorcenters — APPROVED** [review]

**Comment by phipolis** [comment]
@inode- This branch has merge conflicts with develop, probably caused by the Python 3 changes now merged there. Do you have any capacity to resolve these conflicts? I'm terribly excited to see the Bad FCS issue go away. Thanks for this patch!

**Comment by taylorcenters** [comment]
I'll take a shot at adding this patch for the current py3 version of kb in a new PR

**Comment by taylorcenters** [comment]
Please check out #229 for this change made for python3.

I will probably rebuild the apimote firmeware and add it there too - Might be worth trying to set up some github pipeline that will build it and make an artifact available rather than just uploading compiled firmware.

**Comment by taylorcenters** [comment]
Thanks so much for this PR. The change got merged in with #229

**Pull request closed without merging**